### PR TITLE
Chore/GH Actions CD on master

### LINKED ISSUE #382: Move from Travis-CI.org to GitHub Actions
Travis-ci.org is closing down with everything moving over to Travis-ci.com. Generally this would be okay; however it travis-ci.com uses a credit based system (10 credits per minute). Based on some rough estimations we run about 300 minutes (3,000 credits) per build for the Vega Strike Engine Source; master will reduce this some as we have dropped some builds. If Travis-ci.com did a recurring allotment each month we could easily figure out our monthly needs, adjusting the allotment accordingly. That said, it's not clear that the open source credits from Travis-ci.com would apply to GitHub Organizations (still waiting to hear back on an email to support). All said, we'd likely need to request 1,000,000 credit chunk allotments to cover us for a sufficient time period across our repositories. All-in-all it's not as attractive.

Doing some research we have a few options:
- move to Circle-CI (also credit based, don't think they have an open source policy though...so we'd run out of our credits quickly and be stuck each month unless we paid for more credits)
- move to our own infrastructure (we have to pay for our infrastructure but we get unlimited usage...problem is we don't have much income yet to support doing so).
- move to GitHub Actions (free for public repos, otherwise credit based as well from what I can tell at [GH Actions Billing Info](https://docs.github.com/en/free-pro-team@latest/actions/reference/usage-limits-billing-and-administration#about-billing-for-github-actions)); they also have a [migration guide](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/migrating-from-travis-ci-to-github-actions).

GH Actions seems to be where we want to go for now at least. It would be good to identify any "actions" we need to do and survey the [GH Action Marketplace](https://github.com/marketplace?type=actions) to see if there are any that we can utilize or if we need to build our own:
- [ ] CMake Build Runner [GH Actions Search](https://github.com/marketplace?type=actions&query=cmake)
- [ ] CMake Test Runner [GH Actions Search](https://github.com/marketplace?type=actions&query=cmake)
- [ ] CMake Package Runner [GH Actions Search](https://github.com/marketplace?type=actions&query=cmake)
- [ ] Coveralls.IO Integration or other code coverage replacement
- [ ] GitHub Assets Uploader (for releases)

### PR DESCRIPTION
**Pull request by stephengtuggy**
Thank you for submitting a pull request and becoming a contributor to the Vega Strike Core Engine.

Please answer the following:

Code Changes:
- [ ] Have the PR Validation Tests been run? See https://github.com/vegastrike/Vega-Strike-Engine-Source/wiki/Pull-Request-Validation
- [ ] This is a documentation change only
- [X] This is a CI/CD/build system change only

Issues:
- Please list any related issues
Closes #382 

Purpose:
- What is this pull request trying to do? Apply the same CI/CD changes from 0.7.x to master
- What release is this for? 0.8.x
- Is there a project or milestone we should apply this to? 0.8.x?

### CODE REVIEW

File: .github/workflows/gh-actions-release.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - FROM:     'ubuntu:focal'
+            COMPILER: 'gcc'
+            FLAGS:    '-DUSE_PYTHON_3=ON'
+          - FROM:     'ubuntu:focal'
+            COMPILER: 'clang'
```
Comment (BenjamenMeyer) [inline .github/workflows/gh-actions-release.yml:22]: let's capture #400 in here too.

File: README.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 <p style="font-weight:600; font-size:36px">Vega Strike Space Simulation Engine</p>
 </header>
 
-[![Build Status](https://travis-ci.org/vegastrike/Vega-Strike-Engine-Source.svg?branch=master)](https://travis-ci.org/vegastrike/Vega-Strike-Engine-Source)
+<!-- ![CI](https://github.com/vegastrike/Vega-Strike-Engine-Source/workflows/gh-actions-pr/badge.svg) -->
```
Comment (BenjamenMeyer) [inline README.md:5]: this will probably become a list of all the gh-actions we do that we deem relevant; but this is a good start.

**Review by BenjamenMeyer — CHANGES_REQUESTED** [review]
Let's capture #400  in here too before we merge this.

File: .github/workflows/gh-actions-release.yml (hunk 1 above)
Comment (stephengtuggy) [reply .github/workflows/gh-actions-release.yml:19]: Oops! Fixed. I don't know how I missed that earlier

**Review by BenjamenMeyer — APPROVED** [review]

**Pull request merged**